services/src/main/java/org/keycloak/protocol/oidc/DefaultTokenExchangeProvider.java

```diff
@@ -311,10 +313,16 @@ protected Response exchangeClientToClient(UserModel targetUser, UserSessionModel
             }
         }
 
-        if (targetClient.isConsentRequired()) {
-            event.detail(Details.REASON, "audience requires consent");
-            event.error(Errors.CONSENT_DENIED);
-            throw new CorsErrorResponseException(cors, OAuthErrorException.INVALID_CLIENT, "Client requires user consent", Response.Status.BAD_REQUEST);
+        // verification of existing users consents for requested scopes of target client
+        if (!TokenManager.verifyConsentStillAvailable(session, targetUser, targetClient, TokenManager.getRequestedClientScopes(OAuth2Constants.SCOPE, client))) {
```

Comment:
Is it really passing over the requested client scopes? Looks like you are only passing the const `OAuth2Constants.SCOPE` to `TokenManager.getRequestedClientScopes`?

The requested scopes are available from the following lines.